exchange/exchange.go

```diff
@@ -581,15 +582,27 @@ func applyCategoryMapping(ctx context.Context, requestExt openrtb_ext.ExtRequest
 			}
 
 			var categoryDuration string
+			var dupeKey string
 			if brandCatExt.WithCategory {
 				categoryDuration = fmt.Sprintf("%s_%s_%ds", pb, category, newDur)
+				dupeKey = category
 			} else {
 				categoryDuration = fmt.Sprintf("%s_%ds", pb, newDur)
+				dupeKey = categoryDuration
 			}
 
-			if dupe, ok := dedupe[categoryDuration]; ok {
+			if dupe, ok := dedupe[dupeKey]; ok {
 				// 50% chance for either bid with duplicate categoryDuration values to be kept
-				if rand.Intn(100) < 50 {
+				dupeBidPrice, dupeBidPriceErr := strconv.ParseFloat(dupe.bidPrice, 64)
```

Comment:
Nitpick: The variables name seem a bit long. Consider shortening to `dupePrice` and `err`.

Reply:
Fixed!